test/integration/protocol_integration_test.cc

```diff
@@ -1656,7 +1651,10 @@ TEST_P(ProtocolIntegrationTest, 304WithBody) {
     ASSERT_FALSE(response->complete());
   }
 
-  upstream_request_->encodeData(2, true);
+  // Note that at this point the parser is expecting a new response, and it
+  // signals an error because it cannot parse the received data as a valid
+  // status line.
+  upstream_request_->encodeData("aa\r\n", true);
```

Comment:
I think this makes sense and is reasonable, but I also wonder if this kind of diff (IIUC where the legacy parser can signal an error right away, and Balsa needs the \r\n) may come back to bite at some point. I'm guessing this diff is present for a general scenario of an invalid response that doesn't have \r\n, so I think it makes sense for this change to go through and consider the general scenario separately. (And maybe the result of that consideration is that this is NBD.) WDYT? Thanks!

Reply:
Yes, this is definitely worth keeping in mind.  This difference has already come up, for example, at https://github.com/envoyproxy/envoy/pull/22525/files.  Reworking Balsa would requiring not only some engineering work, but also every functional change faces tremendous operational inertia in Google's deployment.  My plan is to proceed with this current behavioral change, and revisit only if a business need arises from Envoy's side.

Reply:
Makes sense, thanks!

Reply:
+1 on looking at this holistically in #21245